pennylane/circuit_drawer/mpldrawer.py

```diff
@@ -0,0 +1,465 @@
+            line = plt.Line2D((-1, self.n_layers), (wire, wire), zorder=1)
+            self.ax.add_line(line)
+
+    def label(self, labels):
+        """Label each wire.
+
+        Args:
+            labels [Iterable[str]]: Iterable of labels for the wires
+
+        **Example**
+
+        .. code-block:: python
+
+            drawer = MPLDrawer(n_wires=2, n_layers=1)
+            drawer.label(["a", "b"])
+
+        .. figure:: ../../_static/drawer/labels.png
+            :align: center
+            :width: 60%
+            :target: javascript:void(0);
+
+        """
+        for wire, ii_label in enumerate(labels):
+            self.ax.text(-1.5, wire, ii_label)
+
+    def box_gate(
+        self, layer, wires, text="", extra_width=0, rotate_text=False, zorder_base=0, color=None
+    ):
+        """Draws a box and adds label text to it's center.
+
+        Args:
+            layer (Int)
+            wires (Union[Int, Iterable[Int]])
+            text (str)
+
+        Kwargs:
+            extra_width=0 (float): Extra box width
+            rotate_text=False (Bool): whether to rotate text 90 degrees. Helpful to long labels and
+                multi-wire boxes.
+            zorder_base (Int): shift the object in zorder
+            color=None: mpl compatible color designation
+
+        **Example**
+
+        .. code-block:: python
+
+            drawer = MPLDrawer(n_wires=2, n_layers=2)
+
+            drawer.box_gate(layer=0, wires=0, text="Y")
+            drawer.box_gate(layer=1, wires=(0,1), text="CRy(0.1)", rotate_text=True)
+
+        .. figure:: ../../_static/drawer/box_gates.png
+            :align: center
+            :width: 60%
+            :target: javascript:void(0);
+
+        """
+        wires = _to_tuple(wires)
+
+        box_min = min(wires)
+        box_max = max(wires)
+        box_len = box_max - box_min
+        box_center = (box_max + box_min) / 2.0
+
+        if rotate_text:
+            rotation = "vertical"
+        else:
+            rotation = "horizontal"
+
+        box = plt.Rectangle(
+            (layer - self._box_dx - extra_width / 2, box_min - self._box_dx),
+            2 * self._box_dx + extra_width,
+            (box_len + 2 * self._box_dx),
+            zorder=2 + zorder_base,
+            facecolor=color,
+        )
+        self.ax.add_patch(box)
+
+        self.ax.text(
+            layer,
+            box_center,
+            text,
+            zorder=3 + zorder_base,
+            rotation=rotation,
+            ha="center",
+            va="center",
+            fontsize="x-large",
+        )
+
+    def ctrl(self, layer, wire_ctrl, wire_target=tuple(), color=None):
+        """Add an arbitrary number of control wires
+
+        Args:
+            layer (Int): the layer to draw the object in
+            wire_ctrl (Union[Int, Iterable[Int]]): set of wires to control on
+            wire_target=tuple() (Union[Int, Iterable[Int]]): target wires. Used to determine min
```

Comment:
```suggestion
            wire_target (tuple(Union[Int, Iterable[Int]])): target wires. Used to determine min
```
This occurs a few places in the docstrings, I'm not sure how to specify it (@josh146 will know :+1:)

Reply:
🤔 I think it normally is
```rest
wire_target (tuple[Union[Int, Iterable[Int]]]): target wires. Used to determine min
```
although there is also the option to use `or` instead of Union (sphinx should understand it as a keyword). This can sometimes be more readable:

```rest
wire_target (tuple[Int or Iterable[Int]]): target wires. Used to determine min
```